Sources/IntegerUtilities/LCM.swift

```diff
@@ -0,0 +1,35 @@
+//===--- LCM.swift --------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// The least common multiple of a list of values.
+///
+/// TODO
+///
+/// [wiki]: https://en.wikipedia.org/wiki/Least_common_multiple
+@inlinable
+public func lcm<T: BinaryInteger>(_ n: T...) -> T {
+    n.reduce(1, _lcm(_:_:))
+}
```

Comment:
I am assuming there aren't a lot of algorithms based around a variadic input and these functions are rather useful in a context where the number of parameter is known at the time of calling these functions: In this case I wonder if it is worth to skip the empty set entirely, since it does look like an error and we instead enforce at least one value.

```suggestion
@inlinable
public func lcm<T: BinaryInteger>(_ a: T, _ n: T...) -> T {
    _lcm(a, n.reduce(1, _lcm(_:_:)))
}
```

The same applies to the variadic `gcd`, leaving the empty set to `Sequence` based implementations for both of these functions instead (which I expect will be used by algorithms that build upon either of the two).

Reply:
I am fine with this and non-empty enforcement.